discovery-client/src/main/java/io/micronaut/discovery/vault/config/client/filter/VaultHttpClientFilter.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.micronaut.discovery.vault.config.client.filter;
+
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.annotation.Value;
+import io.micronaut.discovery.vault.VaultClientConstants;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.MutableHttpRequest;
+import io.micronaut.http.annotation.Filter;
+import io.micronaut.http.filter.ClientFilterChain;
+import io.micronaut.http.filter.HttpClientFilter;
+import org.reactivestreams.Publisher;
+
+/**
+ *  Vault Http Filter used to set auth token header.
+ *
+ *  @author thiagolocatelli
+ *  @author graemerocher
+ *  @since 1.1.1
+ */
+@Filter("/v1/**")
```

Comment:
This URI seems too broad. Does this logic need implementing with a filter?

Reply:
I use the filter to include the Vault authentication token, and their api starts with `/v1` but depending on the version of the KV, what comes after changes.

Reply:
@thiagolocatelli Can it not just be done through the client itself instead of a filter?

Reply:
@jameskleeh if there is another way to include a header that is not using a filter, I need to look at the documentation. Any place I should be looking at to get an example?

Reply:
Just add an argument to the client method `@Header("X-Vault-Token") String token`

Reply:
@jameskleeh that easy! Thank you, sir.